src/Orleans.Runtime/Hosting/CoreHostingExtensions.cs

```diff
@@ -92,17 +93,38 @@ public static ISiloHostBuilder ConfigureLocalHostPrimarySilo(this ISiloHostBuild
         /// <summary>
         /// Configure silo to use GrainBasedMembership
         /// </summary>
-        public static ISiloHostBuilder UseGrainBasedMembership(this ISiloHostBuilder builder)
+        public static ISiloHostBuilder UseGrainBasedMembership(this ISiloHostBuilder builder, Action<GrainBasedMembershipOptions> configureOptions)
         {
-            return builder.ConfigureServices(services => services.UseGrainBasedMembership());
+            return builder.ConfigureServices(services => services.UseGrainBasedMembership(configureOptions));
+        }
+
+        /// <summary>
+        /// Configure silo to use GrainBasedMembership
+        /// </summary>
+        public static ISiloHostBuilder UseGrainBasedMembership(this ISiloHostBuilder builder, Action<OptionsBuilder<GrainBasedMembershipOptions>> configureOptions)
```

Comment:
UseGrainBasedClustering? Maybe there's a friendlier name which also indicates that it shouldn't really be used in production when resilience is important, like UseDevelopmentClustering?

Reply:
Yes, I think I would prefer something more obvious than grain based. Nevertheless, I understand we are changing the nuget packages for membership and gateway list providers to be in a "Clustering" package, but I though this change didn't affect the implementations?
If I use Clustering for what we currently have for membership, how do we disambiguate with gateway list providers?

Reply:
I'd say that whatever we decide regarding Clustering vs Membership, we should do it in a different pass.
Still, I'd like to address the other part of this feedback in this PR, should I call it `UseDevelopmentMembership`? `UseNonReliableMembership`?
I think for transactions we already started calling some things Development, but the extension method was kept as `UseInMemoryTransactionLog()`.
What do you guys say we should do here?

Reply:
In transaction, there's a namespace `Orleans.Transactions.Development` contains stuff we recommended to use in Development, and also added in xml comments of "Recommended for Development only"

It is a bit weird to add intended use to the method name. It differs from our naming convention on membership too. All of our membership is named with the technology used behind it.

Reply:
Whatever we choose, I hope is implies this to developers that this is inherently not fault tolerant and is primarily intended for dev/test scenarios.

If we say `UseInMemoryClustering()`, they might assume it's in-memory but replicated (which is valid and relatively fault-tolerant). If we say `UseDevelopmentClustering()` there's the chance they will be scared to use it even in cases where they have a fixed set of on-prem servers. I'm *ok* with that.

Maybe someone else has a suggestion?

Reply:
Yes, I agree with Reuben, this is not similar to the InMemory stuff we have even. So I'll go with `UseDevelopmentMembership` for now (that we can then rename to `UseDevelopmentClustering` if that's what we decide for all providers)